add missing func_reference.c to raw_sources_list in CMakeLists.txt
fixes "error: undefined reference to 'xc_func_reference_get_ref'"
cfr. https://gitlab.com/libxc/libxc/issues/91 + https://gitlab.com/libxc/libxc/merge_requests/165
--- libxc-4.3.4/CMakeLists.txt.orig	2019-06-07 11:47:56.377984881 +0200
+++ libxc-4.3.4/CMakeLists.txt	2019-06-07 11:48:17.508415975 +0200
@@ -77,6 +77,7 @@
 bessel.c
 expint_e1.c
 func_info.c
+func_reference.c
 functionals.c
 gga.c
 gga_c_am05.c
